GitHub Actions で flake8 の Lint をする

flake8 で PEP 8 に準拠しているかをチェックする設定を VS Code に入れました。 #82 

参考: 以前 GitHub Actions にいれた black の Lint #78 

現状では警告が多すぎるので、差し当たり GitHub Actions では `python -m flake8 --exit-zero --max-line-length=120 --exclude .venv` で PEP 8 に準拠してない箇所が発見されても、 `--exit-zero` で正常終了ながら結果は出る、みたいな状態にしておこうと思っています。

開発していくうちに解消できたら `--exit-zero` を取り除いてクリーンにして CI で検出するようにします。

現在のチェック結果です。

```
$ flake8  --max-line-length=120 --exclude .venv
./setup.py:1:1: F401 'os' imported but unused
./budgetmapper/serializers.py:2:1: F401 'rest_framework.fields' imported but unused
./budgetmapper/serializers.py:2:1: F401 'rest_framework.validators' imported but unused
./budgetmapper/serializers.py:22:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:28:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:36:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:53:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:59:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:67:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:73:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:92:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:112:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:134:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:143:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:149:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:155:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:174:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:188:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:202:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/serializers.py:224:9: F811 redefinition of unused 'fields' from line 2
./budgetmapper/views.py:4:1: F401 'django_filters.rest_framework as drf_filters' imported but unused
./budgetmapper/views.py:7:1: F401 'django_filters.rest_framework as filters' imported but unused
./budgetmapper/views.py:9:1: F401 'rest_framework.exceptions' imported but unused
./budgetmapper/tests/test_request.py:65:9: F841 local variable 'bud' is assigned to but never used
./budgetmapper/tests/test_request.py:66:31: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:70:32: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:115:9: F841 local variable 'abi000' is assigned to but never used
./budgetmapper/tests/test_request.py:116:9: F841 local variable 'abi001' is assigned to but never used
./budgetmapper/tests/test_request.py:117:9: F841 local variable 'abi002' is assigned to but never used
./budgetmapper/tests/test_request.py:118:9: F841 local variable 'abi010' is assigned to but never used
./budgetmapper/tests/test_request.py:119:9: F841 local variable 'abi100' is assigned to but never used
./budgetmapper/tests/test_request.py:290:32: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:300:36: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:321:36: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:421:32: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:431:36: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:451:36: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:567:31: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:662:32: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:674:36: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:706:36: F541 f-string is missing placeholders
./budgetmapper/tests/test_request.py:945:9: F841 local variable 'abi00' is assigned to but never used
./budgetmapper/tests/test_request.py:946:9: F841 local variable 'abi01' is assigned to but never used
./budgetmapper/tests/test_request.py:947:9: F841 local variable 'abi10' is assigned to but never used
./budgetmapper/tests/test_request.py:948:9: F841 local variable 'abi11' is assigned to but never used
./budgetmapper/tests/test_request.py:949:9: F841 local variable 'abi12' is assigned to but never used
./budgetmapper/tests/test_request.py:1003:9: F841 local variable 'abi00' is assigned to but never used
./budgetmapper/tests/test_request.py:1004:9: F841 local variable 'abi01' is assigned to but never used
./budgetmapper/tests/test_request.py:1005:9: F841 local variable 'abi10' is assigned to but never used
./budgetmapper/tests/test_request.py:1006:9: F841 local variable 'abi11' is assigned to but never used
./budgetmapper/tests/test_request.py:1007:9: F841 local variable 'abi12' is assigned to but never used
./budgetmapper/tests/test_request.py:1095:9: F841 local variable 'abi00' is assigned to but never used
./budgetmapper/tests/test_request.py:1096:9: F841 local variable 'abi01' is assigned to but never used
./budgetmapper/tests/test_request.py:1114:9: F841 local variable 'abi00' is assigned to but never used
./budgetmapper/tests/test_request.py:1115:9: F841 local variable 'abi01' is assigned to but never used
./budgetmapper/tests/test_request.py:1149:5: F811 redefinition of unused 'test_destroy_requires_login' from line 1144
./budgetmapper/tests/test_serializer.py:1:1: F401 'unittest.mock.MagicMock' imported but unused
./budgetmapper/tests/test_serializer.py:1:1: F401 'unittest.mock.patch' imported but unused
```

--- a/.github/workflows/django.yml
+++ b/.github/workflows/django.yml
@@ -33,10 +33,14 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -e .[dev]
-      - name: Lint
+      - name: Lint with black
         working-directory: ./backend
         run: |
           python -m black --check --diff --line-length 120 --skip-string-normalization .
+      - name: Lint with flake8
+        working-directory: ./backend
+        run: |
+          python -m flake8 --exit-zero --max-line-length=120 --exclude .venv
       - name: Setup
         working-directory: ./backend
         run: |

--- a/.github/workflows/django.yml
+++ b/.github/workflows/django.yml
@@ -33,11 +33,11 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -e .[dev]
-      - name: Lint with black
+      - name: Prettier
         working-directory: ./backend
         run: |
           python -m black --check --diff --line-length 120 --skip-string-normalization .
-      - name: Lint with flake8
+      - name: Lint
         working-directory: ./backend
         run: |
           python -m flake8 --exit-zero --exclude .venv --max-line-length=120

--- a/.github/workflows/django.yml
+++ b/.github/workflows/django.yml
@@ -40,7 +40,7 @@ jobs:
       - name: Lint with flake8
         working-directory: ./backend
         run: |
-          python -m flake8 --exit-zero --max-line-length=120 --exclude .venv
+          python -m flake8 --exit-zero --exclude .venv --max-line-length=120
       - name: Setup
         working-directory: ./backend
         run: |

